txnsync/peer.go

```diff
@@ -368,6 +372,9 @@ func (p *Peer) selectPendingTransactions(pendingTransactions []pooldata.SignedTx
 	var effectiveBloomFilters []int
 	effectiveBloomFilters = make([]int, 0, len(p.recentIncomingBloomFilters))
 	for filterIdx := len(p.recentIncomingBloomFilters) - 1; filterIdx >= 0; filterIdx-- {
+		if p.recentIncomingBloomFilters[filterIdx].filter == nil {
```

Comment:
we might want to limit the size of the `effectiveBloomFilters` so it would not be bigger than X ( not sure what the X should be.. )